### DISCUSSION TITLE
Want to resolve files without adding each time index.tsx at the end

### DISCUSSION THREAD

**Discussion by andriiradkevych** [freshframework/fresh#2066] [General]
`import { withStyles } from 'hoc/withStyles/index.tsx';` 

Above you can see one of my general imports, I want to make it shorter and make it be resolvable by writting :

`import { withStyles } from 'hoc/withStyles';`

is it possible to configure ?

**Comment by marvinhagemeister** [comment]
You can add an entry for that in the import map inside `deno.json`:

```json
{
  "imports": {
    "hoc/withStyles": "hoc/withStyles.index.tsx"
  }
}
```

That said it sounds more like you want to apply this to every file for which the import map workaround certainly is not sufficient. The automatic mapping to `index` files is something that's only available in CommonJS or when bundlers are used. ESM by definition requires you to specify the full file path, including the filename and extension.

**Comment by lfjsa** [comment]
I know this is not _strictly_ what you asked, but you can skip the folder/index.ts convention in favour of just the file.

`import { withStyles } from 'hoc/withStyles.tsx'`

Deno does not treat `index.ts|tsx` files differently ([styleguide](https://docs.deno.com/runtime/manual/references/contributing/style_guide#do-not-use-the-filename-indextsindexjs)) and a flat structure works perfectly fine ([stdlib src](https://deno.land/std@0.208.0/collections?source)).

**Reply by andriiradkevych** [reply]
yeah, thanks , seems as good solution
